pallets/author-mapping/src/lib.rs

```diff
@@ -234,11 +311,16 @@ pub mod pallet {
 		OptionQuery,
 	>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn account_deposit_keys_of)]
+	pub type MappingWithDepositAndKeys<T: Config> =
```

Comment:
The most important point of my review: we should not keep the old storage but replace it with the new one.
We can change the type of the value for map `MappingWithDeposit` without migration with a custom implementation of Decode for `RegistrationInfo`.

Reply:
For examble, something like that:

```rust
impl<AccountId: Decode, Balance: Decode>  Decode for RegistrationInfo<AccountId, Balance> {
	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
			if let Ok((account, deposit))  = <(AccountId, Balance)>::decode(input) {
				Ok(RegistrationInfo { account, deposit, keys: None })
			} else {
				let (account, deposit, keys) = <(AccountId, Balance, Option<Keys>)>::decode(input)?;
				Ok(RegistrationInfo { account, deposit, keys })
		}
	}
}

Reply:
this example implies that we add the field `keys: Option<Keys>`

Reply:
The advantage of this approach is that we don't need to modify the storage (so no migration), and we keep the existing map.
When we need the VRF key, it's enough to say that if it's None we create it from the NimbusId.

Reply:
We decided to go with the migration instead which assigns the keys to the nimbus id.